Userland/Libraries/LibFileSystemAccessClient/Client.h

```diff
@@ -32,6 +32,9 @@ class Client final
     DeprecatedResult try_open_file(GUI::Window* parent_window, DeprecatedString const& window_title = {}, StringView path = Core::StandardPaths::home_directory(), Core::OpenMode requested_access = Core::OpenMode::ReadOnly);
     DeprecatedResult try_save_file_deprecated(GUI::Window* parent_window, DeprecatedString const& name, DeprecatedString const ext, Core::OpenMode requested_access = Core::OpenMode::WriteOnly | Core::OpenMode::Truncate);
 
+    // FIXME: We should take Core::Stream::OpenMode when both server and client side use only Core::Stream
```

Comment:
Is there anything that is keeping us from just doing that now? The new API already has a `Core::Stream::File` return type, why couldn't it take a `Core::Stream::OpenMode` parameter?

Granted, the current IPC protocol requires `Core::OpenMode`, but we might as well just translate those inside of `try_save_file` to avoid leaking parts of the old API into the new API.

Judging by the following code in `request_file_handler`, we only need to handle `ReadOnly` and `WriteOnly` anyways, and their combination of `ReadWrite`:

```
auto relevant_permissions = requested_access & (Core::OpenMode::ReadOnly | Core::OpenMode::WriteOnly);
```